src/sdb.c

```diff
@@ -764,14 +778,18 @@ SDB_API bool sdb_foreach(Sdb* s, SdbForeachCallback cb, void *user) {
 	if (!result) {
 		return sdb_foreach_end (s, false);
 	}
+
 	ut32 i;
-	for (i = 0; i < s->ht->size; i++) {
-		ls_foreach_safe (s->ht->table[i], iter, tmp, kv) {
-			if (!kv || !sdbkv_value (kv) || !*sdbkv_value (kv)) {
-				continue;
-			}
-			if (!cb (user, sdbkv_key (kv), sdbkv_value (kv))) {
-				return sdb_foreach_end (s, false);
+	for (i = 0; i < s->ht->size; ++i) {
+		HtBucket *bt = &s->ht->table[i];
+		SdbKv *kv;
+		ut32 j;
+
+		BUCKET_FOREACH (s->ht, bt, j, kv) {
```

Comment:
FIXME: make sure this work even when `cb` deletes the current element (AFAIK this happens in radare2 code)

Reply:
BUCKET_FOREACH_SAFE?